cg/meta/upload/gt.py

```diff
@@ -43,93 +47,142 @@ def data(self, analysis: Analysis) -> dict:
-                case_obj=analysis.case, hk_version=hk_version
-            )
+            gt_data["samples_sex"] = self._get_samples_sex_mip_dna(case=analysis.case)
+        elif analysis.workflow == Workflow.RAREDISEASE:
+            gt_data["samples_sex"] = self._get_samples_sex_raredisease(case=analysis.case)
         else:
             raise ValueError(f"Workflow {analysis.workflow} does not support Genotype upload")
-        return data
-
-    def _get_samples_sex_mip(self, case_obj: Case, hk_version: Version) -> dict:
-        qc_metrics_file = self.get_qcmetrics_file(hk_version)
-        analysis_sexes = self.analysis_sex(qc_metrics_file)
-        samples_sex = {}
-        for link_obj in case_obj.links:
-            sample_id = link_obj.sample.internal_id
+        return gt_data
+
+    def upload(self, data: dict, replace: bool = False):
+        """Upload data about genotypes for a family of samples."""
+        self.gt.upload(str(data["bcf"]), data["samples_sex"], force=replace)
+
+    @staticmethod
+    def _is_suitable_for_genotype_upload(case: Case) -> bool:
+        """Check if a cancer case is contains WGS and normal sample."""
+
+        samples: list[Sample] = case.samples
+        return any(
+            (not sample.is_tumour and PrepCategory.WHOLE_GENOME_SEQUENCING == sample.prep_category)
+            for sample in samples
+        )
+
+    def _get_genotype_file(self, case_id: str) -> File:
+        tags: set[str] = {GenotypeAnalysisTag.GENOTYPE}
+        hk_genotype = self.hk.get_file_from_latest_version(bundle_name=case_id, tags=tags)
+        if not hk_genotype:
+            raise FileNotFoundError(f"Genotype file not found for {case_id}")
+        LOG.debug(f"Found genotype file {hk_genotype.full_path}")
+        return hk_genotype
+
+    def _get_samples_sex_balsamic(self, case: Case) -> dict[str, dict[str, str]]:
+        """Return sex information from StatusDB and from analysis prediction (UNKNOWN for BALSAMIC)."""
+        samples_sex: dict[str, dict[str, str]] = {}
+        for case_sample in case.links:
+            if case_sample.sample.is_tumour:
+                continue
+            sample_id: str = case_sample.sample.internal_id
             samples_sex[sample_id] = {
-                "pedigree": link_obj.sample.sex,
-                "analysis": analysis_sexes[sample_id],
+                "pedigree": case_sample.sample.sex,
+                "analysis": Sex.UNKNOWN,
             }
         return samples_sex
 
-    def _get_samples_sex_balsamic(self, case_obj: Case) -> dict:
-        samples_sex = {}
-        for link_obj in case_obj.links:
-            if link_obj.sample.is_tumour:
-                continue
-            sample_id = link_obj.sample.internal_id
+    def _get_samples_sex_mip_dna(self, case: Case) -> dict[str, dict[str, str]]:
+        """Return sex information from StatusDB and from analysis prediction (stored Housekeeper QC metrics file)."""
```

Comment:
```suggestion
        """Return sex information from StatusDB and from analysis prediction"""
```
Implementation details should not be part of doc str